pkgs/development/tools/rust/cargo-leptos/default.nix

```diff
@@ -0,0 +1,31 @@
+{ stdenv, rustPlatform , fetchFromGitHub, Security, lib, pkg-config, openssl }:
+rustPlatform.buildRustPackage rec {
+  pname = "cargo-leptos";
+  version = "0.1.8";
+  buildFeatures = [ "no_downloads" ]; # cargo-leptos will try to download Ruby and other things without this feature
+
+  src = fetchFromGitHub {
+    owner = "leptos-rs";
+    repo = pname;
+    rev = version;
+    hash = "sha256-z4AqxvKu9E8GGMj6jNUAAWeqoE/j+6NoAEZWeNZ+1BA=";
+  };
+
+  cargoSha256 = "sha256-w/9W4DXbh4G5DZ8IGUz4nN3LEjHhL7HgybHqODMFzHw=";
```

Comment:
You should be able to use `cargoLock.lockFile = "${src}/Cargo.lock";` instead of having to manually provide the hash since all the dependencies come from crates.io. I'm not sure if hydra is going to be fond of that however.